server/src/main/java/org/elasticsearch/repositories/ShardSnapshotInfo.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.snapshots.SnapshotInfo;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ShardSnapshotInfo implements Writeable {
+    private final IndexId indexId;
+    private final SnapshotInfo snapshotInfo;
+    private final ShardId shardId;
+    private final String indexMetadataIdentifier;
+    @Nullable
+    private final String shardStateIdentifier;
+
+    public ShardSnapshotInfo(
+        IndexId indexId,
+        ShardId shardId,
+        SnapshotInfo snapshotInfo,
+        String indexMetadataIdentifier,
+        @Nullable String shardStateIdentifier
+    ) {
+        assert snapshotInfo.indices().contains(indexId.getName());
+
+        this.indexId = indexId;
+        this.shardId = shardId;
+        this.snapshotInfo = snapshotInfo;
```

Comment:
AFAICT this is the complete `SnapshotInfo`, including all indices and shard failures (including stack traces) and so on. That might be pretty sizeable, and mostly not relevant to the shard in question. Can we just pick out the bits we need?